terraform/Makefile

```diff
@@ -27,12 +27,12 @@ endif
 
 YQVERSION=$(shell yq --version | $(SED) -e 's/^yq //' -e 's/^.*version //')
 YQMAJOR=$(shell YQV="$(YQVERSION)"; echo "$${YQV%%.*}")
-ifeq ($(YQMAJOR),2)
-  YQ=yq --yaml-output
-  YQIN=
-else
+ifeq ($(YQMAJOR),4)
   YQ=yq eval
   YQIN=-
+else
```

Comment:
I tried it with yq in version 2.4.1, and also 3.4.1 but I am still getting `unknown flag: --yaml-output`

Reply:
Looks like we need to differentiate differently between the two yq variants out there.

Reply:
There are also two potential sources of `yq`:

[mikefarah/yq](https://github.com/mikefarah/yq) - can be downloaded as binary or installed via snap
- I don't see `--yaml-output` flag
- I see `eval` command in the *v4*

[kislyuk/yq](https://github.com/kislyuk/yq) - installed with pip and it has **jq** dependency
- I see `--yaml-output` flag
- I don't see `eval` command

Reply:
I tested the current version and I think that it is sufficient.
```bash
ifeq ($(YQMAJOR),v4)
  YQ=yq eval
  YQIN=-
else
  YQ=yq --yaml-output
  YQIN=
endif
```
When the [mikefarah/yq](https://github.com/mikefarah/yq) is used, we fall into the **ifeq v4** case, because the output of *--version* looks like this `yq (https://github.com/mikefarah/yq/) version v4.33.3`
When the [kislyuk/yq](https://github.com/kislyuk/yq) is used, we fall into the **else** case, because the output of *--version* looks like this `yq 3.2.2`

If we want to support [mikefarah/yq](https://github.com/mikefarah/yq) **< v4**, we need to introduce more complexity to this and it also means that we will support deprecated and unmaintained tools, because [v2](https://mikefarah.gitbook.io/yq/v/v2.x/) and also [v3](https://mikefarah.gitbook.io/yq/v/v3.x/) is deprecated and unsupported.

What do you think @jschoone @garloff?

Reply:
Hi @chess-knight, thanks for picking this up again.
This looks good. I tested a bit with switching between [mikefarah/yq](https://github.com/mikefarah/yq) v4.33 and [kislyuk/yq](https://github.com/kislyuk/yq) 3.2.2 and it all worked.
I also agree to not support the old versions of [mikefarah/yq](https://github.com/mikefarah/yq)